### PR TITLE
imapsync: 1.727 -> 2.200

### PR DESCRIPTION
**Pull request by elohmeier**
###### Description of changes

Updated imapsync to the latest version and added missing Perl dependencies. Supersedes #128462.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/imapsync/default.nix [hunk 1]
```diff
@@ -19,18 +27,44 @@ stdenv.mkDerivation rec {
...
+    UnicodeString
+    perl
   ];
 
   meta = with lib; {
-    homepage = "http://www.linux-france.org/prj/imapsync/";
+    homepage = "https://imapsync.lamiral.info/";
     description = "Mail folder synchronizer between IMAP servers";
-    license = licenses.gpl2Plus;
-    platforms = platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/imapsync/default.nix:33]: Please set platforms again.

Comment (elohmeier) [reply pkgs/tools/networking/imapsync/default.nix:33]: imapsync supports Darwin (see https://imapsync.lamiral.info/INSTALL.d/INSTALL.Darwin.txt), that's why I've remove that line. It's unfortunately currently blocked by perlPackages.CryptOpenSSLRandom being broken on Darwin (seems similar to python3Packages.pyopenssl to me). Should we add `broken = stdenv.isDarwin;`?

Comment (SuperSandro2000) [reply pkgs/tools/networking/imapsync/default.nix:33]: Then we should add ``platforms = platforms.unix;`` and yes please mark ``perlPackages.CryptOpenSSLRandom`` broken on darwin

Comment (elohmeier) [reply pkgs/tools/networking/imapsync/default.nix:33]: Thanks, done. perlPackages.CryptOpenSSLRandom is already marked as broken on Darwin.

**Review by pSub — APPROVED** [review]

**Comment by SebTM** [comment]
I wanted to upstream an updated done myself and used for one migration (imap-to-imap) when I noticed your PR. I tried to review / merge it with my changes but I'm unable to get it building on "x86_64-linux".

`
bsMF42 | "/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/bin/perl" -Mblib=.. run_with_inc.pl ./boot -q -B -Oparl
bsMF42 | ./boot: extraction of /build/Tr9PXHN15u/par-6e6978626c64/cache-10e91eb0fcd0ac81118024db75ad927b8fd58d98//nix/store/bzd91shky9j9d43girrrj6vmqlw7x9m8-glibc-2.35-163/lib/ld-linux-x86-64.so.2 failed (errno=2)
bsMF42 | system(./boot -I/nix/store/qsrzwhdn2jwpnx9mv89n8496dxzmk9q8-perl5.36.0-PAR-Packer-1.056/lib/perl5/site_perl -I/nix/store/g0zhf66c9p7lgsb2bii192ag53qsdb2a-perl5.36.0-PAR-Dist-0.51/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/g0zhf66c9p7lgsb2bii192ag53qsdb2a-perl5.36.0-PAR-Dist-0.51/lib/perl5/site_perl/5.36.0 -I/nix/store/g0zhf66c9p7lgsb2bii192ag53qsdb2a-perl5.36.0-PAR-Dist-0.51/lib/perl5/site_perl -I/nix/store/9b5i6l3ryi191qjml9n2xkrd28fagp7r-perl5.36.0-PAR-1.017/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/9b5i6l3ryi191qjml9n2xkrd28fagp7r-perl5.36.0-PAR-1.017/lib/perl5/site_perl/5.36.0 -I/nix/store/9b5i6l3ryi191qjml9n2xkrd28fagp7r-perl5.36.0-PAR-1.017/lib/perl5/site_perl -I/nix/store/gz31mqagnqbz261f9cj59l538zzg8yaz-perl5.36.0-Module-ScanDeps-1.29/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/gz31mqagnqbz261f9cj59l538zzg8yaz-perl5.36.0-Module-ScanDeps-1.29/lib/perl5/site_perl/5.36.0 -I/nix/store/gz31mqagnqbz261f9cj59l538zzg8yaz-perl5.36.0-Module-ScanDeps-1.29/lib/perl5/site_perl -I/nix/store/rpp42zmk6sq5hzgz9z9xy0x6ab834sqd-perl5.36.0-IPC-Run3-0.048/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/rpp42zmk6sq5hzgz9z9xy0x6ab834sqd-perl5.36.0-IPC-Run3-0.048/lib/perl5/site_perl/5.36.0 -I/nix/store/rpp42zmk6sq5hzgz9z9xy0x6ab834sqd-perl5.36.0-IPC-Run3-0.048/lib/perl5/site_perl -I/nix/store/gf9kfkssspikw04gfhkj51lb20rvdpsd-perl5.36.0-Getopt-ArgvFile-1.11/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/gf9kfkssspikw04gfhkj51lb20rvdpsd-perl5.36.0-Getopt-ArgvFile-1.11/lib/perl5/site_perl/5.36.0 -I/nix/store/gf9kfkssspikw04gfhkj51lb20rvdpsd-perl5.36.0-Getopt-ArgvFile-1.11/lib/perl5/site_perl -I/nix/store/bsy7n44y14fjd823bkx0gmycbm08j2x3-perl5.36.0-Archive-Zip-1.68/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/bsy7n44y14fjd823bkx0gmycbm08j2x3-perl5.36.0-Archive-Zip-1.68/lib/perl5/site_perl/5.36.0 -I/nix/store/bsy7n44y14fjd823bkx0gmycbm08j2x3-perl5.36.0-Archive-Zip-1.68/lib/perl5/site_perl -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl/5.36.0 -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl -I../blib/arch -I../blib/lib -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl/5.36.0 -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl -I/nix/store/bsy7n44y14fjd823bkx0gmycbm08j2x3-perl5.36.0-Archive-Zip-1.68/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/bsy7n44y14fjd823bkx0gmycbm08j2x3-perl5.36.0-Archive-Zip-1.68/lib/perl5/site_perl/5.36.0 -I/nix/store/bsy7n44y14fjd823bkx0gmycbm08j2x3-perl5.36.0-Archive-Zip-1.68/lib/perl5/site_perl -I/nix/store/gf9kfkssspikw04gfhkj51lb20rvdpsd-perl5.36.0-Getopt-ArgvFile-1.11/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/gf9kfkssspikw04gfhkj51lb20rvdpsd-perl5.36.0-Getopt-ArgvFile-1.11/lib/perl5/site_perl/5.36.0 -I/nix/store/gf9kfkssspikw04gfhkj51lb20rvdpsd-perl5.36.0-Getopt-ArgvFile-1.11/lib/perl5/site_perl -I/nix/store/rpp42zmk6sq5hzgz9z9xy0x6ab834sqd-perl5.36.0-IPC-Run3-0.048/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/rpp42zmk6sq5hzgz9z9xy0x6ab834sqd-perl5.36.0-IPC-Run3-0.048/lib/perl5/site_perl/5.36.0 -I/nix/store/rpp42zmk6sq5hzgz9z9xy0x6ab834sqd-perl5.36.0-IPC-Run3-0.048/lib/perl5/site_perl -I/nix/store/gz31mqagnqbz261f9cj59l538zzg8yaz-perl5.36.0-Module-ScanDeps-1.29/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/gz31mqagnqbz261f9cj59l538zzg8yaz-perl5.36.0-Module-ScanDeps-1.29/lib/perl5/site_perl/5.36.0 -I/nix/store/gz31mqagnqbz261f9cj59l538zzg8yaz-perl5.36.0-Module-ScanDeps-1.29/lib/perl5/site_perl -I/nix/store/9b5i6l3ryi191qjml9n2xkrd28fagp7r-perl5.36.0-PAR-1.017/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/9b5i6l3ryi191qjml9n2xkrd28fagp7r-perl5.36.0-PAR-1.017/lib/perl5/site_perl/5.36.0 -I/nix/store/9b5i6l3ryi191qjml9n2xkrd28fagp7r-perl5.36.0-PAR-1.017/lib/perl5/site_perl -I/nix/store/g0zhf66c9p7lgsb2bii192ag53qsdb2a-perl5.36.0-PAR-Dist-0.51/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/g0zhf66c9p7lgsb2bii192ag53qsdb2a-perl5.36.0-PAR-Dist-0.51/lib/perl5/site_perl/5.36.0 -I/nix/store/g0zhf66c9p7lgsb2bii192ag53qsdb2a-perl5.36.0-PAR-Dist-0.51/lib/perl5/site_perl -I/nix/store/qsrzwhdn2jwpnx9mv89n8496dxzmk9q8-perl5.36.0-PAR-Packer-1.056/lib/perl5/site_perl -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl/5.36.0 -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/5.36.0/x86_64-linux-thread-multi -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/5.36.0 -I. -q -B -Oparl) failed: $?=65280
bsMF42 | make[1]: *** [Makefile:890: parl] Error 255
bsMF42 | make[1]: Leaving directory '/build/PAR-Packer-1.056/myldr'
bsMF42 | make: *** [Makefile:573: subdirs] Error 2
bsMF42 | error: builder for '/nix/store/qxx1ciyajasx28fbsab24qcc7pp7wywb-perl5.36.0-PAR-Packer-1.056.drv' failed with exit code 2
`
(Full build log: https://privatebin.net/?89d85db95939b1b8#FSGv4JZ9HjuboBRh3gg2ATLao4RN9uj8XKUXCMpwDnTv)

I used a similar changes except the PAR*-dependencies, are they optional or what are the required for? Do you have an idea how to solve this? (after that we could update to 227)

File: pkgs/tools/networking/imapsync/default.nix [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
-    url = "https://releases.pagure.org/imapsync/imapsync-${version}.tgz";
-    sha256 = "1axacjw2wyaphczfw3kfmi5cl83fyr8nb207nks40fxkbs8q5dlr";
+  src = fetchFromGitHub {
+    owner = "imapsync";
+    repo = "imapsync";
+    rev = "imapsync-${version}";
+    sha256 = "sha256-EM8nT9v6ZHpEG33/ibPqvfgFoAF36nhq6Y+RUWK4vR0=";
   };
 
   patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/imapsync/default.nix:20]:
```suggestion
  postPatch = ''
```

Comment (elohmeier) [reply pkgs/tools/networking/imapsync/default.nix]: Fixed, thanks.

**Review by Ma27 — CHANGES_REQUESTED** [review]
This doesn't evaluate right now for me:

```
❯ ~/Projects/nixpkgs-prs imapsync → nix-build -A imapsync
error: undefined variable 'LWPUserAgent'

       at /home/ma27/Projects/nixpkgs-prs/pkgs/tools/networking/imapsync/default.nix:42:5:

           41|     JSONWebToken
           42|     LWPUserAgent
             |     ^
           43|     MailIMAPClient
(use '--show-trace' to show detailed location information)

```

Note: this is only reproducible with `allowAliases = false;`.

**Comment by Ma27** [comment]
OK, so `PARPacker` fails to run on x86_64-linux (hence the ofborg error). I managed to fix that by adding `-O0`, i.e. disabling all of gcc's optimizations. This is a strong indicator for undefined behavior in this package's C code and thus I'm strongly against adding this as-is.

**Comment by Ma27** [comment]
The other PR appears to skip the broken PARPacker and apparently it's still working fine, so I'd be in favor of getting #128462 ready.

**Comment by elohmeier** [comment]
@Ma27 Thanks for your review. I've fixed the alias issue and removed the PARPacker dependency.

**Review by SebTM — APPROVED** [review]
Thanks for adjusting, checked out your PR and tested it locally and works fine now :)

**Review by eliasp — APPROVED** [review]

**Comment by eliasp** [comment]
Result of `nixpkgs-review pr 190558` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>imapsync</li>
    <li>perl534Packages.EncodeIMAPUTF7</li>
    <li>perl536Packages.EncodeIMAPUTF7</li>
  </ul>
</details>

**Review by SebTM — APPROVED** [review]

**Comment by SebTM** [comment]
Build and tested again locally (x86_64-linux)

**Pull request merged**